storage/client/src/main/java/com/azure/storage/blob/BlockBlobAsyncClient.java

```diff
@@ -134,33 +152,77 @@ public static BlockBlobClientBuilder builder() {
      *         {@link Metadata}
      * @param accessConditions
      *         {@link BlobAccessConditions}
-     * @param context
-     *         {@code Context} offers a means of passing arbitrary data (key/value pairs) to an
-     *         {@link com.azure.core.http.HttpPipeline}'s policy objects. Most applications do not need to pass
-     *         arbitrary data to the pipeline and can pass {@code Context.NONE} or {@code null}. Each context object is
-     *         immutable. The {@code withContext} with data method creates a new {@code Context} object that refers to
-     *         its parent, forming a linked list.
      *
      * @return
      *      A reactive response containing the information of the uploaded block blob.
      */
-    public Mono<Void> upload(Flux<ByteBuffer> data, long length, BlobHTTPHeaders headers,
-            Metadata metadata, BlobAccessConditions accessConditions, Context context) {
+    public Mono<Response<BlockBlobItem>> upload(Flux<ByteBuffer> data, long length, BlobHTTPHeaders headers,
+            Metadata metadata, BlobAccessConditions accessConditions) {
         return blockBlobAsyncRawClient
-            .upload(data.map(Unpooled::wrappedBuffer), length, headers, metadata, accessConditions, context)
-            .then();
+            .upload(data.map(Unpooled::wrappedBuffer), length, headers, metadata, accessConditions)
+            .map(rb -> new SimpleResponse<>(rb, new BlockBlobItem(rb.deserializedHeaders())));
     }
 
     public Mono<Void> uploadFromFile(String filePath) {
-        return this.uploadFromFile(filePath, null, null, null, null);
+        return this.uploadFromFile(filePath, BLOB_DEFAULT_UPLOAD_BLOCK_SIZE, null, null, null);
     }
 
-    public Mono<Void> uploadFromFile(String filePath, BlobHTTPHeaders headers, Metadata metadata,
-            BlobAccessConditions accessConditions, Context context) {
-        //TODO make this method smart
-        return this.blockBlobAsyncRawClient
-            .upload(ByteBufFlux.fromPath(Paths.get(filePath)), new File(filePath).length())
-            .then();
+    public Mono<Void> uploadFromFile(String filePath, Integer blockSize, BlobHTTPHeaders headers, Metadata metadata,
```

Comment:
I thought I left a comment (though I can't find it now) that we should have these on the Raw client as well. in addition, I noticed CommonRestResponse was deleted. What do you guys think should be the return type for this on the raw client so that customers can access all the headers (because it could either be a putBlob or putBlockList)? Or is the argument that the raw client only has methods that mirror the rest api so we don't have convenience things like uploadFromFile

Reply:
The raw clients are hidden, I thought. The customer can't interact with them and (I believe) they will be removed before GA for the logic to be re-integrated into the async client. There doesn't seem to be a reason to put these methods on there and it feels like more to maintain.

Reply:
Agree with James.